docs/app/(examples)/pattern/filter/utils.tsx

```diff
@@ -0,0 +1,87 @@
+import { venueTypes, venues } from '@/lib/data/venues';
+import { parseAsJson, useQueryState } from 'nuqs';
+import { z } from 'zod';
+import { NumericFormat } from '@marigold/system';
+
+// URL
+// ---------------
+const urlSchema = z.object({
+  type: z.number().optional(),
+  capacity: z.number().optional(),
+  price: z.number().optional(),
+  rating: z.number().optional(),
+});
+
+export type VenueFilter = z.infer<typeof urlSchema>;
+
+export const defaultFilter = {
+  type: undefined,
+  capacity: Math.max(...venues.map(venue => venue.capacity)),
+  price: Math.max(...venues.map(venue => venue.price.to)),
+  rating: undefined,
+} satisfies VenueFilter;
+
+// Form
+// ---------------
+// TODO: is this correct? are there no numbers???
```

Comment:
in zod there are numbers: https://zod.dev/?id=numbers I think you could directly use it for capacity and price for example

Reply:
in `urlSchema` you also used `z.number()` maybe I don't get it 🙈

Reply:
this is all strings because this is how we get the data in the form submission. because form values are all strings.